### ISSUE TITLE
Which sol file should I use to verify and publish my smart contract on Etherscan.io?

### ISSUE THREAD

**Issue by xirkle** [oraclesorg/ico-wizard#375]
Hi!

I have all of these ".sol" files from the ICO wizard. https://github.com/xirkle/crowdsale

Which sol file should I use to verify and publish my smart contract on Etherscan.io?

Thanks.

**Comment by xirkle** [comment]
I tried to verify "003_CrowdsaleTokenExt" and I get the following error message:

Sorry! The Compiled Contract ByteCode for 'Xirkle' does NOT match the Contract Creation Code for [0x8d083b6984b57b0a7b5a196cb68d6159288d1493].

Contract name(s) found: 'CrowdsaleTokenExt' , 'ERC20' , 'ERC20Basic' , 'MintableTokenExt' , 'Ownable' , 'ReleasableToken' , 'SafeMath' , 'SafeMathLibExt' , 'StandardToken' , 'UpgradeAgent' , 'UpgradeableToken' 
Unable to Verify Contract at this point time.

**Comment by vbaranov** [comment]
@xirkle could you post the screenshot of filled inputs from Etherscan verification page?

**Comment by igorbarinov** [comment]
@xirkle please follow a guide about verification of smart contracts 

https://github.com/hashguide/ico-wizard/wiki/ICO-Wizard-Verifying-Contracts

**Comment by xirkle** [comment]
Yes, I already read those instructions at https://github.com/hashguide/ico-wizard/wiki/ICO-Wizard-Verifying-Contracts and I think I'm doing it correctly unless I'm mistaken.

By the way, I ran the codes from "003_CrowdsaleTokenExt.sol" in Remix Solidity and it also produced this new error: 

**browser/xirkle1.sol:35:3: Warning: No visibility specified. Defaulting to "public".
  function Ownable() {
  ^
Spanning multiple lines.**

**Comment by vbaranov** [comment]
@xirkle it is not a error - it is warning. It doesn't interrupt to compile with Remix.

**Comment by xirkle** [comment]
Screen for inputs on Etherscan: https://www.screencast.com/t/5Gi3GhkAVJjJ

**Comment by vbaranov** [comment]
It is wrong contract name in your screenshot. Please, use specific contract names from instruction: https://github.com/hashguide/ico-wizard/wiki/ICO-Wizard-Verifying-Contracts, depending on what contract you are trying to verify.

**Comment by xirkle** [comment]
OK. I'm little confused now.

My presale is live: https://wizard.oracles.org/crowdsale?addr=0x8d083b6984b57B0a7b5A196Cb68D6159288D1493&networkID=1

I'm trying to verify Crowdsale Contract Address: 0x8d083b6984b57B0a7b5A196Cb68D6159288D1493
Is this correct?

When I completed the ICO wizard, I downloaded and extracted the following files:
https://www.screencast.com/t/ZjtJPKxho

Which file(s) should I verify on Etherscan.io?

Thank you!
